Makefile

```diff
@@ -1,13 +1,23 @@
 # this is licensed software, @see LICENSE file.
+
+# OpenCV & Tensorflow recommended flags for performance..
 CFLAGS = -Ofast -march=native -fno-trapping-math -fassociative-math -funsafe-math-optimizations -Wall -pthread
 LDFLAGS = -lrt -ldl
 
+# Version
+VERSION=$(shell git describe --all --long --always --dirty)
+ifeq ($(VERSION),)
+	VERSION=v0.2.0-no-git
+endif
+
+CFLAGS += -D DEEPSEG_VERSION=$(VERSION)
```

Comment:
This puts the version in the define without the quotes. Also prone to blanks in the tag name. Should both be quoted in the makefile and for the actual program source, so as to skip on the `__STR` macro trick.

Reply:
git does not permit tags, branches or other references with white space or escape characters in (https://git-scm.com/docs/git-check-ref-format), so I consider this safe without quoting. I often find that attempting to quote things at indeterminate levels of make/shell/compiler wrapper etc. usually results in pain, hence I try to avoid the need.